module3/src/com/epam/arrayofcharacters/BusinessWorker.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.arrayofcharacters;
+
+
+public class BusinessWorker {
+    //task 1
+    public static String[] convertTitlesToSnakeCase(String[] camelCase) {
+        String[] snakeCase = new String[camelCase.length];
+        for (int i = 0; i < snakeCase.length; i++) {
+            StringBuilder stringBuilder = new StringBuilder();
+            for (int j = 0; j < camelCase[i].length(); j++) {
+                char letter = camelCase[i].toCharArray()[j];
+                if (Character.isLowerCase(letter)) {
+                    stringBuilder.append(letter);
+                } else {
+                    stringBuilder.append("_").append(Character.toLowerCase(letter));
+                }
+            }
+            snakeCase[i] = stringBuilder.toString();
+        }
+        return snakeCase;
+    }
+
+    //task 2
+    public static String replaceTheWords(String text) {
+        StringBuilder strBuilder = new StringBuilder();
+        for (int i = 0; i < text.length(); i++) {
+            if (i + 3 < text.length() && text.charAt(i) == 'w' && text.charAt(i + 1) == 'o' && text.charAt(i + 2) == 'r'
+                    && text.charAt(i + 3) == 'd') {
+                strBuilder.append("letter");
+                i += 3;
+            } else {
+                strBuilder.append(text.charAt(i));
+            }
+        }
+        return strBuilder.toString();
+    }
+
+    //task 3
+    public static int countDigits(String text) {
+        int count = 0;
+        final char [] characters = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
+        for (int i = 0; i < text.length(); i++) {
+            for (char element : characters) {
+                if (text.charAt(i) == element) {
+                    count++;
+                }
+            }
+        }
+        return count;
+    }
```

Comment:
I think that this way is more efficient:
private static final int ASCII_NUMBER_OF_FIRST_DIGIT = 48;
    private static final int ASCII_NUMBER_OF_LAST_DIGIT = 57;

    // task 03
    public static int countDigits(String str) {
        int counterOfDigits = 0;
        final char[] characters = str.toCharArray();
        for (char ch : characters) {
            if (ch >= ASCII_NUMBER_OF_FIRST_DIGIT && ch <= ASCII_NUMBER_OF_LAST_DIGIT) {
                counterOfDigits++;
            }
        }
        return counterOfDigits;
    }